playerbot/strategy/values/AttackersValue.cpp

```diff
@@ -116,64 +106,98 @@ void AttackersValue::RemoveNonThreating(set<Unit*>& targets)
     }
 }
 
-bool AttackersValue::IsPossibleTarget(Unit *attacker, Player *bot, float range)
+bool AttackersValue::IsPossibleTarget(Unit *attacker, Player *player, float range)
 {
+    if(!attacker)
+        return false;
+
     Creature *c = dynamic_cast<Creature*>(attacker);
+    Group* group = player->GetGroup();
+    Player* master = nullptr;
 
     bool rti = false;
-    if (attacker && bot->GetGroup())
-        rti = bot->GetGroup()->GetTargetIcon(7) == attacker->GetObjectGuid();
-
-    PlayerbotAI* ai = bot->GetPlayerbotAI();
-    
     bool leaderHasThreat = false;
-    if (attacker && bot->GetGroup() && ai->GetMaster())
-        leaderHasThreat = attacker->getThreatManager().getThreat(ai->GetMaster());
-
     bool isMemberBotGroup = false;
-    if (bot->GetGroup() && ai->GetMaster() && ai->GetMaster()->GetPlayerbotAI() && !ai->GetMaster()->GetPlayerbotAI()->IsRealPlayer())
-        isMemberBotGroup = true;
+    bool inVehicle = false;
+    bool hasEnemyPlayerTarget = false;
+    bool inDuel = false;
+    bool canSeeAttacker = false;
+    bool hasShackleUndeadAura = false;
+    bool hasAttackTaggedStrategy = false;
+    bool isSapped = false;
+    bool isGouged = false;
+    bool isStunned = false;
+    bool isPolymorphed = false;
+    bool isFeared = false;
+    bool isFriendly = false;
+    bool isDead = false;
+
+    // If the player is a bot
+    PlayerbotAI* bot = player->GetPlayerbotAI();
+    if(bot)

```

Comment:
This method can be called from players that are not bots too, so we need to check it